src/api/dependency-graph/src/router.js

```diff
@@ -0,0 +1,13 @@
+const { Router } = require('@senecacdot/satellite');
+
+module.exports = function (graphConstructor) {
+  const graph = graphConstructor();
+
+  const router = Router();
+
+  router.get('/project/:id', async (req, res) => {
```

Comment:
There's no `await` in this function, do you need `async`?

Reply:
I am sure that I don't need it for the router callback, but I am not sure how to resolve the `async` chain that I have with the whole `graph-collector` module. I just wanted to use `async` because of the `fetch` request, and then it got out of control

Reply:
All you need to worry about is adding a `try/catch` here, as well as `next` to deal with any promise rejections:

```js
router.get('/projects/:id', async (req, res, next) = {
  const { id } = req.params;
  try {
    res.status(200).json(await graph.getProject(id));
  } catch(err) {
    next(err);
  }
});
```

Also note: for a REST API endpoint, we always use plural forms: `/projects` vs. `/project`